### PR TITLE
Incorporate Seeding of Randoms in Stress and Pairwise Generation

### LINKED ISSUE #5732: Support Seeding Randoms in Stress Test

### PR DESCRIPTION
**Pull request by anthony-murphy**
This should make our stress runs more repeatable, but allow more variation per run, as we also seed the options generation, so each run should get a unique set of options, as a bonus, using a random in the pairwise generator also makes the generation more efficient, specifically fewer instance to cover the same number of pairs.

fixes #5732

### CODE REVIEW

File: packages/test/test-pairwise-generator/src/test/generatePairwiseOptions.spec.ts [hunk 1]
```diff
@@ -108,15 +108,15 @@ function  validatePairsExhaustively<T>(
 
 describe("generatePairwiseOptions",()=>{
     it("SimpleOptions",()=>{
-        assert.strictEqual(simpleValues.length, 10, optionsToString(...simpleValues));
+        assert.strictEqual(simpleValues.length, 8, optionsToString(...simpleValues));
```
Comment (anthony-murphy) [inline packages/test/test-pairwise-generator/src/test/generatePairwiseOptions.spec.ts:111]: less is more when it comes to pairwise. it means we are covering more pairs with fewer values, so we can more efficiently explore the space

**Comment by anthony-murphy** [comment]
ping @agarwal-navin and @markfields

File: packages/test/test-pairwise-generator/src/index.ts [hunk 2]
```diff
@@ -30,21 +34,22 @@ function applyPairToPartial<T extends Record<string, any>>(
             && partial[pair.jKey] === pair.jVal) {
             return;
         }
-
-        if(found === undefined) {
+        // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+        if(partial.__paritalKeyCount! < keyCount) {
```
Comment (markfields) [inline packages/test/test-pairwise-generator/src/index.ts:38]: Typo - parital

File: packages/test/test-pairwise-generator/src/index.ts [hunk 3]
```diff
@@ -55,7 +60,16 @@ function applyPairToPartial<T extends Record<string, any>>(
     }
 }
 
-export function generatePairwiseOptions<T extends Record<string, any>>(optionsMatrix: OptionsMatrix<T>): T[] {
+export function generatePairwiseOptions<T extends Record<string, any>>(
+    optionsMatrix: OptionsMatrix<T>,
+    randomSeed: number = 0x35843): T[] {
+    const randEng = random.engines.mt19937();
+    if(randomSeed !== undefined) {
```
Comment (markfields) [inline packages/test/test-pairwise-generator/src/index.ts:67]: Always true right?

File: packages/test/test-pairwise-generator/src/index.ts [hunk 4]
```diff
@@ -81,8 +99,7 @@ export function generatePairwiseOptions<T extends Record<string, any>>(optionsMa
         if(partial.__paritalKeyCount !== matrixKeys.length) {
             for(const key of matrixKeys) {
                 if(!(key in partial)) {
```
Comment (markfields) [inline packages/test/test-pairwise-generator/src/index.ts:83]: This logic seems to assume that partials keys are a subset of the matrix keys. Any concerns here? Worth asserting that they have the same number of keys at the end?

File: packages/test/test-service-load/src/loadTestDataStore.ts [hunk 5]
```diff
@@ -281,7 +283,7 @@ class LoadTestDataStore extends DataObject implements ILoadTest {
                         await wait(cycleMs / 2);
                     }else{
                         // Random jitter of +- 50% of opWaitMs
-                        await wait(opsGapMs + opsGapMs * (Math.random() - 0.5));
+                        await wait(opsGapMs + opsGapMs * random.real(0,.5,true)(config.randEng));
```
Comment (markfields) [inline packages/test/test-service-load/src/loadTestDataStore.ts:286]: Ooh a curried function!

File: packages/test/test-service-load/src/nodeStressTest.ts [hunk 6]
```diff
@@ -36,36 +38,44 @@ async function main() {
...
+    const runnerArgs: string[][] = [];
     for (let i = 0; i < profile.numClients; i++) {
         const childArgs: string[] = [
             "./dist/runner.js",
             "--driver", driver,
             "--profile", profile.name,
             "--runId", i.toString(),
-            "--url", url];
+            "--url", url,
+            "--seed", `0x${seed.toString(16)}`,
```
Comment (markfields) [inline packages/test/test-service-load/src/nodeStressTest.ts:77]: Woah I didn't know you could pass in a numerical param as hex and NS would treat it as a number

File: packages/test/test-service-load/src/nodeStressTest.ts [hunk 7]
```diff
@@ -74,15 +84,18 @@ async function orchestratorProcess(
...
-            childArgs,
-            { stdio: "inherit" },
-        );
-        p.push(new Promise((resolve) => process.on("close", resolve)));
+        console.log(childArgs.join(" "));
+        runnerArgs.push(childArgs);
     }
     try{
-    await Promise.all(p);
+        await Promise.all(runnerArgs.map(async (childArgs)=>{
```
Comment (markfields) [inline packages/test/test-service-load/src/nodeStressTest.ts:91]: What's the reason for this little refactor?

File: packages/test/test-service-load/src/runner.ts [hunk 8]
```diff
@@ -71,40 +86,52 @@ async function runnerProcess(
...
-        let counter = 0;
 
-        while(!done) {
-            counter++;
+        let done = false;
+        for(let iteration = 0; !done; iteration++) {
             // Switch between creating new factory vs. reusing factory.
             // Certain behavior (like driver caches) are per factory instance, and by reusing it we hit those code paths
             // At the same time we want to test newly created factory.
-            const factory = counter % 1 === 0 ?
```
Comment (markfields) [inline packages/test/test-service-load/src/runner.ts:89]: Ha whoops

**Review by markfields — APPROVED** [review]
🎂 LGTM

File: packages/test/test-pairwise-generator/src/index.ts (hunk 4 above)
Comment (anthony-murphy) [reply packages/test/test-pairwise-generator/src/index.ts:83]: i'm not concerned. tests will catch it if there is a problem. rather not pay the cost of the assert

File: packages/test/test-service-load/src/nodeStressTest.ts (hunk 7 above)
Comment (anthony-murphy) [reply packages/test/test-service-load/src/nodeStressTest.ts:91]: to print the runner command lines to make it easier to debug, and i didn't want them interleaved with output, which makes it hard to read

**Pull request merged**